cluster/src/main/java/org/apache/zeppelin/interpreter/SnappydataSqlZeppelinInterpreter.java

```diff
@@ -0,0 +1,235 @@
+    String id = contextInterpreter.getParagraphId();
+    String propertyKey = getPropertyKey(cmd);
+    if (null != propertyKey && !propertyKey.equals(DEFAULT_KEY)) {
+      cmd = cmd.substring(propertyKey.length() + 2);
+    }
+    cmd = cmd.trim();
+    if (cmd.trim().startsWith("approxResultFirst=true")) {
+      cmd = cmd.replaceFirst("approxResultFirst=true", "");
+
+
+      if (firstTime) {
+        firstTime = false;
+
+        for (InterpreterContextRunner r : contextInterpreter.getRunners()) {
+          if (id.equals(r.getParagraphId())) {
+
+
+            final InterpreterResult res = executeSql(propertyKey, cmd, contextInterpreter);
+
+            queue.add(r);
+
+            return res;
+          }
+        }
+
+      } else {
+        firstTime = true;
+        /*
+         * try { Thread.sleep(10000); } catch (InterruptedException e) { e.printStackTrace(); }
+         */
+        String query = cmd.replaceAll("with error .*", "");
+        logger.info("Executing cmd " + query);
+
+        /*
+         * contextInterpreter.getAngularObjectRegistry().add("title", "Accurate result",
+         * contextInterpreter.getNoteId(), contextInterpreter.getParagraphId());
+         */
+        return executeSql(propertyKey, cmd, contextInterpreter);
+      }
+      return null;
+    } else {
+      return executeSql(propertyKey, cmd, contextInterpreter);
+
+    }
+
+
+  }
+
+  /**
+   * The content of this method are borrowed from JDBC interpreter of apache zeppelin
+   * @param propertyKey
+   * @param sql
+   * @param interpreterContext
+   * @return
+   */
+  private InterpreterResult executeSql(String propertyKey, String sql,
+      InterpreterContext interpreterContext) {
+
+    String userId = interpreterContext.getAuthenticationInfo().getUser();
+
+    try {
+
+      Statement statement = getStatement(propertyKey, userId);
+
+      if (statement == null) {
+        return new InterpreterResult(InterpreterResult.Code.ERROR, "Prefix not found.");
+      }
+      statement.setMaxRows(getMaxResult());
+
+      StringBuilder msg = null;
+      boolean isTableType = false;
+
+      if (containsIgnoreCase(sql, EXPLAIN_PREDICATE)) {
+        msg = new StringBuilder();
+      } else {
+        msg = new StringBuilder(TABLE_MAGIC_TAG);
+        isTableType = true;
+      }
+
+      ResultSet resultSet = null;
+      try {
+
+        boolean isResultSetAvailable = statement.execute(sql);
+
+        if (isResultSetAvailable) {
+          resultSet = statement.getResultSet();
+
+          ResultSetMetaData md = resultSet.getMetaData();
+
+          for (int i = 1; i < md.getColumnCount() + 1; i++) {
+            if (i > 1) {
+              msg.append(TAB);
+            }
+            msg.append(replaceReservedChars(isTableType, md.getColumnName(i)));
+          }
+          msg.append(NEWLINE);
+
+          int displayRowCount = 0;
+          while (resultSet.next() && displayRowCount < getMaxResult()) {
```

Comment:
getMaxResult can be called once before the loop. It will avoid calling a  method multiple times.